crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -213,36 +216,49 @@ impl<'flox> Publish<'flox, NixAnalysis> {
     #[allow(dead_code)] // until consumed by cli
     /// Check whether store paths are substitutable by a given substituter and
     /// return the associated metadata.
+    ///
+    /// If substituter is None, the local store will be used, which is probably
+    /// only useful for testing.
     async fn get_binary_cache_metadata(
         &self,
-        substituter: SubstituterUrl,
+        substituter: Option<SubstituterUrl>,
     ) -> Result<CacheMeta, PublishError> {
-        let mut cache_meta = CacheMeta {
-            cache_url: substituter,
-            narinfo: vec![],
-            _other: BTreeMap::new(),
-        };
         let nix: NixCommandLine = self.flox.nix(Default::default());
-        for store_path in self.analysis()["element"]["store_paths"]
+        let store_paths = self.analysis()["element"]["store_paths"]
             .as_array()
+            // TODO use CatalogEntry and then we don't need to unwrap
             .unwrap()
             .iter()
-            .map(|value| value.as_str().unwrap())
-        {
-            let store_path =
-                StorePath::from_path(store_path).map_err(PublishError::ParseStorePath)?;
-            let path_info_command = runix::command::PathInfo {
-                installables: [store_path.clone().into()].into(),
+            .map(|value| {
+                // TODO use CatalogEntry and then we don't need to unwrap
+                StorePath::from_path(value.as_str().unwrap())
+                    .map_err(PublishError::ParseStorePath)
+                    .map(Installable::StorePath)
+            })
+            .collect::<Result<Vec<Installable>, _>>()?;
+        let path_info_command = runix::command::PathInfo {
+            installables: store_paths.into(),
+            eval: EvaluationArgs {
+                eval_store: Some("auto".to_string().into()),
                 ..Default::default()
-            };
+            },
+            ..Default::default()
+        };
 
-            let narinfos = path_info_command
-                .run_typed(&nix, &Default::default())
-                .map_err(|nix_err| PublishError::PathInfo(store_path, nix_err))
-                .await?;
-            cache_meta.narinfo.extend(narinfos);
-        }
-        Ok(cache_meta)
+        let narinfos = path_info_command
+            .run_typed(&nix, &NixArgs {
+                common: NixCommonArgs {
+                    store: substituter.clone().map(|url| url.to_string().into()),
+                },
+                ..Default::default()
+            })
+            .map_err(PublishError::PathInfo)
+            .await?;
+        Ok(CacheMeta {
+            cache_url: substituter.unwrap_or(SubstituterUrl::parse("file:///nix/store").unwrap()),
```

Comment:
I think we should require this methods to take a final SubstituterUrl.
Also `file:///nix/store` does not give you much information after all and is invalid once the file leaves your machine

Reply:
```suggestion
            cache_url: substituter,
```